backend/src/projekti/kayttoOikeudetManager.ts

```diff
@@ -164,31 +165,40 @@ export class KayttoOikeudetManager {
       const kayttajas = this.kayttajas;
       const account: Kayttaja | undefined = kayttajas.findByEmail(email);
       if (account) {
-        const newVarahenkilo = mergeKayttaja({ tyyppi: KayttajaTyyppi.VARAHENKILO, muokattavissa: false }, account);
-        if (newVarahenkilo) {
-          // Remove existing varahenkilo if it's different
-          const currentVarahenkilo = this.users.filter((aUser) => aUser.tyyppi == KayttajaTyyppi.VARAHENKILO && !aUser.muokattavissa).pop();
-          if (currentVarahenkilo?.email !== email) {
-            remove(this.users, (aUser) => aUser == currentVarahenkilo);
-            if (currentVarahenkilo && currentVarahenkilo.kayttajatunnus === this.kunnanEdustaja) {
-              this.addOldProjektipaallikkoOrVarahenkiloAsRegularUser(currentVarahenkilo);
-              log.warn("Vanha varahenkilö lisätty normaalikäyttäjäksi, koska hän on kunnan edustaja", { currentVarahenkilo });
-            }
+        const newVelhohenkilo = mergeKayttaja(
+          { tyyppi: isAorL(account.uid) ? KayttajaTyyppi.VARAHENKILO : null, muokattavissa: false },
+          account
+        );
```

Comment:
Onko tässä varmasti oikein päin nämä argumentit?

Reply:
eikös ne ole, eli vain A- ja L-käyttäjät voi saada tuon kayttajatyypin VARAHENKILO, ja muut on sitten ns normaaleja käyttäjiä tyypillä null?

Reply:
eikun mergeKayttaja-funktion argumentit. Oikeanpuoleinen ylirkirjoittaa vasemmanpuoleisen käsittääkseni? Ts. jos käyttäjä oli alunperin KayttajaTyyppi.PROJEKTIPAALLIKKO, niin newVelhohenkilo on projari eikä varahenkilö.

Reply:
Kah, mergeKayttajassa ei otetakaan tyyppiä mukaan, joten tämä ok.